Rachio update

Klikini opened the pull request:
## Description:

Update to https://github.com/home-assistant/home-assistant/pull/7600

## Checklist:

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

Klikini committed:
Rachio platform started

Klikini committed:
Rachio platform started

Klikini committed:
Rachio tests

Klikini committed:
Rachio tests

Klikini committed:
detect bad api token

Commit:
Documentation, Code cleanup

Commit:
Docstrings end with a period, log uses %

Commit:
Merge remote-tracking branch 'refs/remotes/home-assistant/dev' into dev

Commit:
Fix arguments, default run time is now 10 minutes

Commit:
Fix typo, remove todo (GH issue exists)

Commit:
Revert polymer submodule commit

Commit:
Use a RachioPy version with SSL cert validation

Commit:
Update requirements

Commit:
Don't double-convert manual run time

https://github.com/home-assistant/home-assistant/pull/7600#issuecomment-305543153

Commit:
Merge remote-tracking branch 'refs/remotes/home-assistant/dev' into dev

balloob committed:
Version bump to 0.46

emlove committed:
Support for renaming ZWave values (#7780)

* Support for renaming ZWave values

* Improve test

balloob committed:
Fix telegram_bot (#7877)

amelchio committed:
[light.lifx] Update aiolifx (#7882)

This makes LIFX Gen3 lights work with the current firmware.

balloob committed:
Fix MQTT camera test (#7878)

balloob committed:
Merge pull request #7866 from home-assistant/release-0-46

0.46

Commit:
use manual_run_secs instead of the now-missing seconds in turn_on

Commit:
ignore unused kwargs

aequitas committed:
Dsmr5 revert (#7900)

* Revert "Update to dsmr_parser supporting v5 arguments."

This reverts commit 3567de4b90361c262c4d3261325c2b7dbfa5441f.

* Revert "Using dev branch until released upstream."

This reverts commit 53e8de112cb0c47ac5031aacd74a5b18b4a418e2.

* Revert "Give good example."

This reverts commit 4f90fc4be677634fc7e951b4b79aa57c8213b8f8.

* Revert "Allow configuring DSMR5 protocol."

This reverts commit 9fa0e14187e32521e2f51cf2183a9070ee093bd0.

mje-nz committed:
Fix typos in Wunderground component (Percipitation -> Precipitation) (#7901)

cribbstechnologies committed:
Mqtt cover modifications (#7841)

* adding set position ability
removing command_topic being required

* flaking

* flaking test

* updating docs

* requested updates

* Revert "updating docs"

This reverts commit 9cfc5ed7a8f5761de93d8bc7c1cfb81fa8d7ca40.

* forgot to update constructor calls in tests

Commit:
use upstream RachioPy, fix manual run switches

Commit:
merge dev

balloob commented:
Please create a new branch instead of reusing your old branch.

balloob commented on homeassistant/components/switch/rachio.py:
```diff
@@ -19,7 +19,7 @@
 DEFAULT_MANUAL_RUN_MINS = 10
 
-MIN_UPDATE_INTERVAL = timedelta(minutes=5)
+MIN_UPDATE_INTERVAL = timedelta(seconds=5)
```
?

Klikini replied:
https://community.home-assistant.io/t/rachio-in-0-46-0/19185/13?u=klikini

balloob replied:
Rachio offers an API for free but hitting it every 5 seconds is going to get us banned.

brg468 replied:
@balloob, I've been running mine with a 30 second interval and it seems to have the same desired effect, in that regardless of 5 or 30 seconds it updates the correct state on the frontend after about 30 seconds have passed.  Do you think leaving timedelta at 30 seconds would be acceptable to not overload the API?

balloob commented on homeassistant/components/switch/rachio.py:
```diff
@@ -30,6 +30,7 @@
 
 
 # noinspection PyUnusedLocal
+# noinspection PyUnresolvedLocal
```
Please remove these warnings. They are not part of linters that the project uses.

Klikini replied:
https://github.com/Klikini/home-assistant/tree/rachio-part2

balloob replied:
I don't understand this comment.

fabaff committed:
Fix changes introduced with #7917 (#7930)

fabaff committed:
Do not call update() in constructor (#7931)

mpstephana committed:
Add service_url config option to volvooncall (#7919)

* Add service_url config option to volvooncall

* Import default value from volvooncall lib

jminn committed:
Added 'change' field to statistics sensor (#7820)

* Added 'change' field to statistics sensor

* Updated statistics sensor test

* Updated statistics sensor test complaint

nkgilley committed:
add ripple sensor (#7935)

fabaff committed:
Add Yahoo! weather platform (#7939)

bjw-s committed:
[media_player.sonos] Send media_stop on turn_off (#7940)

danielperna84 committed:
Updated dependency

danielperna84 committed:
Merge pull request #7948 from danielperna84/HomeMatic

HomeMatic: Updated dependency

joopert committed:
update to 006 (#7945)

vrs01 committed:
Update ping.py (#7944)

Commit:
Add option to display all input sources / Add support for favourite channels / Treat Marantz SR5008 as Denon AVR-X device (#7949)

* add option to display all sources / pushed to version 0.4.3 of library

* Add show_all_sources option for auto discovery too

* change code style for hass

* fix lint

Teagan42 committed:
[WIP] Fix opencv (#7864)

* Updates to opencv image processor

* Remove opencv hub

* Requirements

* Remove extra line

* Fix linting errors

* Indentation

* Requirements

* Linting

* Check for import on platform setup

* Remove opencv requirement

* Linting

* fix style

* fix lint

fabaff committed:
Upgrade coinmarketcap to 3.0.1 (#7951)

heinemml committed:
test connection without needing admin rights (#7947)

SHOW DIAGNOSTICS always needs admin privileges on influxdb. For
the purposes of home-assistant this is too much.
Use 'SHOW SERIES' to have a relatively lightweight query which
only needs READ privileges.

bokub committed:
Fix the negative values bug in history_stats (#7934)

alanfischer committed:
Vera colored light support (#7942)

* Added support for color to vera lights

* Updated requirements

fabaff committed:
Fix typos (#7957)

Add an optional extended description…

commento committed:
Osram lightify, removed double set to the lightify bridge in case of brightness changes (#7662)

* osram lightify removed duplicated set in case of brightness changes

* lightify component: anticipate brightness evaluation to handle unconsidered scenario described in the PR request comments (light turned on with color/temperature)

* Correction for travis ci error:
undefined name 'transition'

fabaff committed:
Add configuration check and use default var names (#7963)

trisk committed:
lock.sesame: Update pysesame, add state attributes (#7953)

* Update pysesame requirement to 0.1.0 to support caching

* Set `available` property based on API enabled status

* Add state attributes for device ID and battery level

balloob committed:
Version bump to 0.46.1

Commit:
Fixed metadata issue (#7932)

Commit:
Fixed metadata issue (#7932)

girlpunk committed:
Create metoffice.py (#7965)

Fix met office sensor

girlpunk committed:
Create metoffice.py (#7965)

Fix met office sensor

balloob committed:
Prevent Roku doing I/O in event loop (#7969)

balloob committed:
Prevent Roku doing I/O in event loop (#7969)

balloob committed:
Merge pull request #7968 from home-assistant/release-0-46-1

0.46.1

balloob committed:
Merge branch 'master' into dev

balloob committed:
Fix platforms being able to block startup (#7970)

balloob committed:
New component: Python Script (#7950)

* Add initial version

* Fix requirements

* Prefer logging over printing

* Set executor thread name on >Py36 only

* Add tests

* Lint

* Add restrictedpython to test dependencies

* Create python_script.py

From doc:
```
However, an empty dict ({}) is treated as is. If you want to specify a list that can contain anything, specify it as dict:
>>> schema = Schema({}, extra=ALLOW_EXTRA)  # don't do this
>>> try:
...   schema({'extra': 1})
...   raise AssertionError('MultipleInvalid not raised')
... except MultipleInvalid as e:
...   exc = e
>>> str(exc) == "not a valid value"
True
>>> schema({})
{}
>>> schema = Schema(dict)  # do this instead
>>> schema({})
{}
>>> schema({'extra': 1})
{'extra': 1}

```

Commit:
Nadtcp component (#7955)

* initial commit

* class name and requirements_all.txt

* removed mentions of D7050

* changed default name

* catch oserror in update, travis errors.

* use nad_receiver pip version

* update coveragerc

danielperna84 committed:
HomeMatic optimizations and code cleanup (#7986)

* Cleanup and optimizations

* Cleanup

* Typo -.-

* Linting

molobrakos committed:
Update eliqonline.py (#7977)

Print error

chilicheech committed:
test that all lights turn off when no entity id is given (#7981)

azogue committed:
make `last_name` field optional (#7988)

Commit:
Bugfixing with version 0.4.4 of denonavr (#7995)

fabaff committed:
Upgrade py-cpuinfo to 3.3.0 (#7992)

amelchio committed:
LIFX: clean up internal color conversions (#7964)

* Add color_util.color_hsv_to_RGB

* Use helper functions for LIFX conversions

The LIFX API uses 16 bits for saturation/brightness while HA uses 8 bits.
Using helper functions makes the conversion a bit nicer and less prone
to off-by-one issues.

The colorsys library uses 0.0-1.0 but we can avoid that by using the HA
color_util converters instead.

amelchio committed:
LIFX: add support for setting infrared level (#8000)

* LIFX: update aiolifx

This adds support for infrared and multizone.

* LIFX: add support for infrared

tedstriker committed:
dismiss service for persistent notifications (#7996)

* dismiss service for persistent notifications

Unnecessary notifications can now be removed automatically. Added a
dismiss service to remove persistent notifications via script and/or
automation.

* removed unnecessary loop

loop removed

fabaff committed:
Add Gitter.im sensor (#7998)

sander76 committed:
Fixing Client connection error (#7991)

emlove committed:
Additional demo fan with only speed support (#7985)

* Additional demo fan with only speed support

* Update demo fan tests

happyleavesaoc committed:
Update mailgun (#7984)

* add mailgun component

* add to coveragerc

soldag committed:
Fixed bug in spotify component. (#7976)

cribbstechnologies committed:
fixing potential null issue with optional param being parsed as a script (#7928)

* fixing potential null issue with optional param being parsed as a script

* Create template.py

fabaff committed:
Make it more flexible (fixes #7954) (#8001)

* Make it more flexible (fixes #7954)

* Fix var name

Sabesto committed:
Add Flexit AC climate platform (#7871)

* Add Flexit AC climate platform

* Protocol extracted to third party lib

nkgilley committed:
update pyripple (#8015)

fabaff committed:
Allow put as method (#8004)

balloob commented:
Please base your changes off a new copy of the dev branch. It now includes a lot of commits that are not part of this changeset.

mjsir911 committed:
Configure conversation for custom actions with keywords (#7734)

* - Simple keyword to action config

* - Added more fuzzy stuff

* - Logging & a bit of commenting

* - pep8?

* - pep8 and quick formatting fixes

* - Changed configuration a bit

* - Backwards compatibility tests

* - Fallback or

* - Added custom configuration for conversation

* - Moved imports inside function

* - pep8

* - Pass tests better

* - Removed unused imports

* - Moved warning ignore to above import for fuzzy

* - Moved return for consistent return types

* - Fallback if no choices to listen for

* - Fixed linting errors

* - Better logging and fixed linting errors(?)

* - Fixed continuation

* - Added one blank line after class docstring

* Create conversation.py

* Create test_conversation.py

* Create test_conversation.py

* Update test_conversation.py

fabaff committed:
Remove globally disabled pylint issues (#8005)

chilicheech committed:
entity_id for service fan.turn_off is optional (#7982)

* entity id is optional

* use a simple if/else to set the data for the fan.turn_off service

Klikini commented:
@balloob: I've never worked with an external organization before. What's the best way to do that?

Commit:
Remove PyCharm linter comments

Commit:
update requirements to use pip version of RachioPy

Commit:
update requirements to use pip version of RachioPy

Commit:
use rachiopy from pip

Commit:
use rachiopy from pip

Commit:
fix merge conflict in Rachio rebase

Commit:
add newline to end of file

balloob commented:
Please create a new PR based off a new branch forked off our dev branch.
